astropy/utils/masked/core.py

```diff
@@ -1055,38 +1055,42 @@ def mean(self, axis=None, dtype=None, out=None, keepdims=False):
                 dtype = np.dtype('f4')
                 is_float16_result = out is None
 
+        where = ~self.mask & where
+
         result = self.sum(axis=axis, dtype=dtype, out=out,
-                          keepdims=keepdims, where=~self.mask)
-        n = np.add.reduce(~self.mask, axis=axis, keepdims=keepdims)
+                          keepdims=keepdims, where=where)
+        n = np.add.reduce(where, axis=axis, keepdims=keepdims)
         result /= n
         if is_float16_result:
             result = result.astype(self.dtype)
         return result
 
-    def var(self, axis=None, dtype=None, out=None, ddof=0, keepdims=False):
+    def var(self, axis=None, dtype=None, out=None, ddof=0, keepdims=False, *, where=True):
+        where = ~self.mask & where
+
         # Simplified implementation based on that in numpy/core/_methods.py
-        n = np.add.reduce(~self.mask, axis=axis, keepdims=keepdims)[...]
+        n = np.add.reduce(where, axis=axis, keepdims=keepdims)[...]
 
         # Cast bool, unsigned int, and int to float64 by default.
         if dtype is None and issubclass(self.dtype.type,
                                         (np.integer, np.bool_)):
             dtype = np.dtype('f8')
-        mean = self.mean(axis=axis, dtype=dtype, keepdims=True)
+        mean = self.mean(axis=axis, dtype=dtype, keepdims=True, where=where)
```

Comment:
@mhvk, do you know why the mask wasn't used on this line before? To my naive understanding, it seems like the mask should be used here.

Reply:
Ok, that makes sense to me now, I needed to reset my brain to think in terms of masked arrays.